lib/scholar/metrics/regression.ex

```diff
@@ -519,6 +519,38 @@ defmodule Scholar.Metrics.Regression do
     Nx.reduce_max(Nx.abs(y_true - y_pred))
   end
 
+  @doc ~S"""
+  Calculates the mean pinball loss to evaluate predictive performance of quantile regression models.
+
+  $$pinball(y, \hat{y}) = \frac{1}{n) \sum_{i=1}^{n} \alpha max(\hat{y_i} - y_i, 0) +
+  (1 - \alpha) max(\hat{y_i} - y_i, 0)$$
+
+  The residual error is defined as $$|y - \hat{y}|$$ where $y$ is a true value
+  and $\hat{y}$ is a predicted value.
+  Equivalent to half of the $$mean_absolute_error$$ when $$\alpha$$ is 0.5.
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([1, 2, 3])
+      iex> y_pred = Nx.tensor([2, 3, 4])
+      iex> Scholar.Metrics.Regression.mean_pinball_loss(y_true, y_pred, 0.5)
+      #Nx.Tensor<
+        f32
+        0.5
+      >
+  """
+  defn mean_pinball_loss(y_true, y_pred, opts \\ []) do
+    check_shape(y_true, y_pred)
+    alpha = opts[:alpha]
+
+    # Formula adapted from sklearn:
+    # https://github.com/scikit-learn/scikit-learn/blob/128e40ed593c57e8b9e57a4109928d58fa8bf359/sklearn/metrics/_regression.py#L299
+    diff = y_true - y_pred
+    sign = diff >= 0
+    loss = alpha * sign * diff - (1 - alpha) * (1 - sign) * diff
+    Nx.mean(loss)
+  end
```

Comment:
Hey this is a total drive by comment, but I actually had to write a mean pinball error a few months ago:

* https://hexdocs.pm/exgboost/quantile_prediction_interval.html#discussion-pinball-error

I mention it because I also looked up sklearn's implementation and found it a bit confusing. I was able to simplify it to this:

```elixir
defmodule Pinball do
  import Nx.Defn

  defn mean_error(y_pred, y_true, alpha) do
    dy = y_true - y_pred
    # if dy < 0, do: alpha, else: alpha - 1
    coeff = alpha + (dy >= 0) - 1
    error = coeff * dy
    Nx.mean(error)
  end
end
```

Feel free to use or disregard :)

Also, I'll be able to update that livebook to use Scholar's version after this PR, so thanks!